### PR TITLE
🐛 fix: add required iam permission for managed node groups

### LINKED ISSUE #2042: Creating a cluster with the `eks-managedmachinepool` flavor fails
/kind bug
/area provider/eks

**What steps did you take and what happened:**
I have created a kind cluster and installed the 0.6.1 version of CAPA. I then create a workload cluster using the `eks-managedmachinepool` flavor using:

```bash
clusterctl config cluster managed-test --flavor eks-managedmachinepools > capi-eks.yaml
kubectl apply -f  capi-eks.yaml
```
The control plane is created correctly but the EKS managed node group is never created and the following can be seen in filt from the controller logs:

```
[manager] E1016 14:16:53.446352      38 controller.go:257] controller-runtime/controller "msg"="Reconciler error" "error"="failed to reconcile machine pool for AWSManagedMachinePool default/capi-managed-test-pool-0: failed to describe nodegroup: failed to describe nodegroup: AccessDeniedException: \n\tstatus code: 403, request id: 89517d88-432c-4b0b-ae72-8abdff36c35a" "controller"="awsmanagedmachinepool" "name"="capi-managed-test-pool-0" "namespace"="default"
```

**What did you expect to happen:**
I expect a managed node group to be created and it be attached to the EKS cluster

**Environment:**

- Cluster-api-provider-aws version: 0.6.1
- Kubernetes version: (use `kubectl version`): 1.17

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:
When creating the bootstrap stack the permissions required for managed machine pools (a.k.a EKS managed machine node groups) aren't added.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2042

### CODE REVIEW

File: pkg/cloud/services/eks/nodegroup.go [hunk 1]
```diff
@@ -49,6 +50,7 @@ func (s *NodegroupService) describeNodegroup() (*eks.Nodegroup, error) {
 			case eks.ErrCodeResourceNotFoundException:
 				return nil, nil
 			default:
+				s.scope.V(2).Info("failed to describe nodegroup", "error", err.Error(), "awserr", aerr)
```
Comment (michaelbeaumont) [inline pkg/cloud/services/eks/nodegroup.go:53]: Should the `.Info()` calls still be here? Do we have any guidelines for when/what to log for these kinds of cases?

Comment (richardcase) [reply pkg/cloud/services/eks/nodegroup.go:53]: I think this one could be removed but the previous one should stay in., I'll remove this one now.

Comment (richardcase) [reply pkg/cloud/services/eks/nodegroup.go]: Removed

**Review by michaelbeaumont — APPROVED** [review]
LGTM

File: cmd/clusterawsadm/cloudformation/bootstrap/managed_control_plane.go [hunk 2]
```diff
@@ -31,5 +31,5 @@ func (t Template) eksControlPlanePolicies() []string {
 }
 
 func eksAssumeRolePolicy() *iamv1.PolicyDocument {
```
Comment (randomvariable) [inline cmd/clusterawsadm/cloudformation/bootstrap/managed_control_plane.go:33]: Maybe give this a more specific function name, or inline it completely.

Comment (richardcase) [reply cmd/clusterawsadm/cloudformation/bootstrap/managed_control_plane.go]: Good idea, this has been inlined.

**Pull request merged**